### DISCUSSION TITLE
connection of two walls

### DISCUSSION THREAD

**Discussion by Theinformatiker** [IfcOpenShell/IfcOpenShell#2816] [Q&A]
How do I make the connection between two wall elements?

```python
# it is only a code sniple :-)
 wall1 = run("root.create_entity", model, ifc_class="IfcWall", name="FirstWall")
 wall2 = run("root.create_entity", model, ifc_class="IfcWall", name="SecoundWall")
representation1 = run("geometry.add_wall_representation", model, context=body, length=5, height=3, thickness=0.2)
representation2 = run("geometry.add_wall_representation", model, context=body, length=5, height=3, thickness=0.2)

run("geometry.assign_representation", model, product=wall1, representation=representation1)
run("geometry.assign_representation", model, product=wall2, representation=representation2)


#a short part of the code


run("geometry.connect_element",model,relating_element = wall1 , related_element = wall2)

```

what is the different of ifcopenshell.api.geometry.connect_element 
and ifcopenshell/api/geometry/connect_path ?  

The next question is whether the connection can be seen correctly in the CAD viewer or elsewhere?

best regards
T.I

**Comment by aothms** [comment]
The difference is in the connection relationship being estabished https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/api/geometry/connect_element.py#L54 https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/api/geometry/connect_path.py#L82

For all wall, connect path is the appropriate way.

> whether the connection can be seen correctly in the CAD viewer or elsewhere?

What do you exactly mean with that. The connections is mostly a logical statement. While there is the potential to have an associated connection geometry, I'm not aware of viewers capable of showing that in case of walls. It's also not being populated at this moment when using the API.

**Reply by Theinformatiker** [reply]
thanks for the explanation
